docs/whatsnew-5.2.rst

```diff
@@ -0,0 +1,373 @@
+and `pyupgrade <https://github.com/asottile/pyupgrade>`_ to assist you with
+this effort.
+
+After the migration is done, run your test suite with Celery 4 to ensure
+nothing has been broken.
+
+Step 5: Upgrade to Celery 5.2
+-----------------------------
+
+At this point you can upgrade your workers and clients with the new version.
+
+.. _v520-important:
+
+Important Notes
+===============
+
+Supported Python Versions
+-------------------------
+
+The supported Python versions are:
+
+- CPython 3.7
+- CPython 3.8
+- CPython 3.9
+- PyPy3.7 7.3 (``pypy3``)
+
+Experimental support
+~~~~~~~~~~~~~~~~~~~~
+
+Celery supports these Python versions provisionally as they are not production
+ready yet:
+
+- CPython 3.10 (currently in RC2)
+
+Memory Leak Fixes
+-----------------
+
+Two severe memory leaks have been fixed in this version:
+
+* :class:`celery.result.ResultSet` no longer holds a circular reference to itself.
+* The prefork pool no longer keeps messages in its cache forever when the master
+  process disconnects from the broker.
+
+The first memory leak occurs when you use :class:`celery.result.ResultSet`.
+Each instance held a promise which provides that instance as an argument to
+the promise's callable.
+This caused a circular reference which kept the ResultSet instance in memory
+forever since the GC couldn't evict it.
+The provided argument is now a :func:`weakref.proxy` of the ResultSet's
+instance.
+The memory leak mainly occurs when you use :class:`celery.result.GroupResult`
+since it inherits from :class:`celery.result.ResultSet` which doesn't get used
+that often.
+
+The second memory leak exists since the inception of the project.
+The prefork pool maintains a cache of the jobs it executes.
+When they are complete, they are evicted from the cache.
+However, when Celery disconnects from the broker, we flush the pool
+and discard the jobs, expecting that they'll be cleared later once the worker
+acknowledges them but that has never been the case.
+Instead, these jobs remain forever in memory.
+We now discard those jobs immediately while flushing.
+
+Dropped support for Python 3.6
+------------------------------
+
+Celery now requires Python 3.7 and above.
+
+Python 3.6 will reach EOL in December, 2021.
+In order to focus our efforts we have dropped support for Python 3.6 in
+this version.
+
+If you still require to run Celery using Python 3.6
+you can still use Celery 5.1.
+However we encourage you to upgrade to a supported Python version since
+no further security patches will be applied for Python 3.6 after
+the 23th of December, 2021.
+
+Tasks
+-----
+
+When replacing a task with another task, we now give an indication of the
+replacing nesting level through the ``replaced_task_nesting`` header.
+
+A task which was never replaced has a ``replaced_task_nesting`` value of 0.
+
+Kombu
+-----
+
+Starting from v5.2, the minimum required version is Kombu 5.2.0.
+
```

Comment:
I miss information about automated killing of orphaned workers in prefork pool: https://github.com/celery/celery/pull/6942

Reply:
```suggestion

Prefork Workers Pool
---------------------

Now all orphaned worker processes are killed automatically when main process exits.

```